opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -126,25 +128,9 @@ class WellModelMatrixAdapter : public Dune::AssembledLinearOperator<M,X,Y>
   WellModelMatrixAdapter (const M& A,
                           const M& A_for_precond,
                           const WellModel& wellMod,
-                          const std::any& parallelInformation OPM_UNUSED_NOMPI = std::any() )
-      : A_( A ), A_for_precond_(A_for_precond), wellMod_( wellMod ), comm_()
-  {
-#if HAVE_MPI
-    if( parallelInformation.type() == typeid(ParallelISTLInformation) )
-    {
-      const ParallelISTLInformation& info =
-          std::any_cast<const ParallelISTLInformation&>( parallelInformation);
-      comm_.reset( new communication_type( info.communicator() ) );
-    }
-#endif
-  }
-  WellModelMatrixAdapter (const M& A,
-                          const M& A_for_precond,
-                          const WellModel& wellMod,
-                          std::shared_ptr<communication_type> comm )
+                          const std::shared_ptr< communication_type >& comm = std::shared_ptr< communication_type >())
```

Comment:
This refactoring does not seem related to the flexible solver? I think it looks better and simpler, so I support the refactoring itself.